zerver/lib/actions.py

```diff
@@ -4949,40 +4949,42 @@ def validate_email_for_realm(target_realm: Realm, email: str) -> None:
             raise AssertionError("Mirror dummy user is already active!")
         # Other users should not already exist at all.
         raise ValidationError(_('%s already has an account') %
-                              (email,), code = _("Already has an account."))
+                              (email,), code = _("Already has an account."), params={'deactivated': False})
     elif not existing_user_profile.is_mirror_dummy:
         raise ValidationError('The account for %s has been deactivated' % (email,),
-                              code = _("Account has been deactivated."))
+                              code = _("Account has been deactivated."), params={'deactivated': True})
 
-def validate_email(user_profile: UserProfile, email: str) -> Tuple[Optional[str], Optional[str]]:
+def validate_email(user_profile: UserProfile, email: str) -> Tuple[Optional[str], Optional[str],
+                                                                   bool]:
     try:
         validators.validate_email(email)
     except ValidationError:
-        return _("Invalid address."), None
+        return _("Invalid address."), None, False
```

Comment:
I don't like the idea of returning `is_deactivated` as a manadatory `return_value` for `validate_email` function. Can you think of another way to solve this?

Reply:
@hackerkid I did try to return it as `Optional[bool]` originally but mypy tests were failing at a later stage ( in function do_invite_users line 5064 and line 5066 argument 1 of append) because of this error `mypy argument incompatible type "Tuple[str, str, Optional[bool]]"; expected "Tuple[str, str, bool]"`.Can you help me out with this?